libraries/chain/include/eosio/chain/trace.hpp

```diff
@@ -19,6 +19,72 @@ namespace eosio::chain {
    struct transaction_trace;
    using transaction_trace_ptr = std::shared_ptr<transaction_trace>;
 
+   // Identify the starting position of a sync call in the call's console log.
+   // This is used for pretty printing console logs.
+   struct console_marker {
+      uint32_t call_ordinal     = 0;
+      uint32_t console_position = 0;
+   };
+
+   struct sync_call_trace {
+      sync_call_trace(uint32_t sender_ordinal, account_name receiver, bool read_only, std::span<const char> data)
+         : sender_ordinal(sender_ordinal)
+         , receiver(receiver)
+         , read_only(read_only)
+         , data(data.begin(), data.end())
+      {
+      }
+
+      // receiver's ordinal. A sequence number starting with 1, unique within
+      // an action. It can be used to reference `call_trace` struct in `call_traces`
+      // vector in `action_trace` struct.
+      uint32_t                      ordinal = 1;
+
+      // sender's ordinal. If the caller is the action, sender_ordinal is 0.
+      const uint32_t                sender_ordinal = 0;
```

Comment:
I should have noticed this sooner -- I think we'd want to make these varints (`fc::unsigned_int`). Ordinals in `action_trace` are also varints.

Reply:
Thanks. Will change.